sources/functions/deluge

```diff
@@ -49,114 +51,204 @@ function remove_ltcheckinstall() {
+            return
+        fi
+    fi
+    false
+    return
+}
+
+function build_libtorrent_deluge() {
+    booststrap
+    echo 'using gcc : : : <cflags>"-march=native -std=c++14" <cxxflags>"-march=native -std=c++14" ;' > /root/user-config.jam
+    echo "using python : ${pythonmajor} : /usr/bin/${pythonver} : /usr/include/python${pythonmajor} : /usr/lib/python${pythonmajor} ;" >> /root/user-config.jam
+    apt_install ${pythonver}-dev
+    rm_if_exists /tmp/libtorrent
+    git clone -b ${libtorrent_branch} https://github.com/arvidn/libtorrent /tmp/libtorrent >> ${log} 2>&1 || {
+        echo_error "Libtorrent could not be cloned from git. Setup will not proceed"
+        exit 1
+    }
+    cd /tmp/libtorrent
+    VERSION=$(grep AC_INIT configure.ac | grep -oP '\d+\.\d+\.\d+')
+
+    if [[ -f /root/libtorrent-${libtorrent_branch}.patch ]]; then
+        patch -p1 < /root/libtorrent-${libtorrent_branch}.patch >> ${log} 2>&1
+    fi
+    #The following sed command is a settings change in the python binding to "force on" force reannounce in the python binding
+    #If not enabled, Deluge may not properly reannounce on unregistered torrents.
+    #For more info on the change and why this is changed (including why this is not default in upstream): https://github.com/arvidn/libtorrent/issues/3226
+    case $libtorrent_branch in
+        RC_1_1)
+            sed -i 's|, (arg("seconds") = 0, arg("tracker_idx") = -1, arg("flags") = 0|, (arg("seconds") = 0, arg("tracker_idx") = -1, arg("flags") = 1|g' bindings/python/src/torrent_handle.cpp
+            #Patch the Jamfile to allow 1.1 to use stage_module
+            patch -p1 < /etc/swizzin/sources/Jamfile-RC_1_1.patch >> ${log} 2>&1
+            ;;
+        RC_1_2)
+            sed -i 's|, (arg("seconds") = 0, arg("tracker_idx") = -1, arg("flags") = reannounce_flags_t{}))|, (arg("seconds") = 0, arg("tracker_idx") = -1, arg("flags") = 1))|g' bindings/python/src/torrent_handle.cpp
+            #sed -i 's|static constexpr reannounce_flags_t ignore_min_interval = 0_bit|static constexpr reannounce_flags_t ignore_min_interval = 1_bit|g' include/libtorrent/torrent_handle.hpp
+            ;;
+        *) ;;
+    esac
+    cd -
+    cd /tmp/libtorrent/bindings/python
+    /opt/boost_${BOOST_VERSION}/b2 -j$(nproc) python=${pythonmajor} crypto=openssl variant=release libtorrent-link=static boost-link=static install_module python-install-path=/tmp/dist/libtorrent-python/usr/local/lib/python${pythonmajor}/dist-packages >> ${log} 2>&1
+    tag=" static with c++14 march=native"
+    mkdir -p /root/dist
+    fpm -f -C /tmp/dist/libtorrent-python -p /root/dist/${python}-libtorrent_VERSION.deb -s dir -t deb -n ${python}-libtorrent --version ${VERSION} --description "Libtorrent rasterbar python binding compiled by swizzin$tag" >> ${log} 2>&1 || {
+        echo_error "Error packaging ${python}-libtorrent. Setup will exit"
+        exit 1
+    }
+    cd -
```

Comment:
What are these `cd -` being used for?

Reply:
`cd -` returns to the previous working directory.

This must be done as you cannot `rm` a folder while located inside of it, thus we must move back to the previous position. Previously, `/tmp` was the directory used for this.

Reply:
What about `cd`ing to `/` instead? making it relative through the use of `-` might be handy but things could go wrong if the previous location no longer exists for whatever reason

Reply:
What was wrong with `/tmp` then?

Reply:
Now that I know this is dependent on these cds, probably nothing. Not sure if it was to temp or a die under it. Sorry about that.